src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/ProPlanControls/ProPlanControls.jsx

```diff
@@ -0,0 +1,107 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useForm } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+
+import {
+  useAccountDetails,
+  useAvailablePlans,
+  usePlanData,
+} from 'services/account'
+import { useProPlans } from 'shared/utils/billing'
+import {
+  getDefaultValuesProUpgrade,
+  getSchema,
+  MIN_NB_SEATS_PRO,
+} from 'shared/utils/upgradeForm'
+import TextInput from 'ui/TextInput'
+
+import BillingControls from './BillingControls'
+import TotalBanner from './TotalBanner'
+import UserCount from './UserCount'
+
+import { useUpgradeControls } from '../hooks'
+import UpdateButton from '../UpdateButton'
+
+function ProPlanControls() {
+  const { provider, owner } = useParams()
+  const { data: accountDetails } = useAccountDetails({ provider, owner })
+  const { data: plans } = useAvailablePlans({ provider, owner })
+  const { data: planData } = usePlanData({ owner, provider })
+  const { proPlanYear } = useProPlans({ plans })
+  const { upgradePlan } = useUpgradeControls()
+
+  const trialStatus = planData?.plan?.trialStatus
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isValid, errors },
+    setValue,
+    getValues,
+  } = useForm({
+    defaultValues: getDefaultValuesProUpgrade({
+      accountDetails,
+      proPlanYear,
+      trialStatus,
+    }),
+    resolver: zodResolver(
+      getSchema({
+        accountDetails,
+        minSeats: MIN_NB_SEATS_PRO,
+        trialStatus,
+      })
+    ),
+    mode: 'onChange',
+  })
+
+  const newPlan = getValues('newPlan')
```

Comment:
A big thing I did was centralize the values we get from the form, either by watching or getting those values here, and passing them where we need to. This way there's one "source of truth" for these